Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1445,7 +1445,7 @@ ExceptionOr<void> HTMLElement::hidePopoverInternal(FocusPreviousElement focusPre
         queuePopoverToggleEventTask(PopoverVisibilityState::Showing, PopoverVisibilityState::Hidden);
 
     if (RefPtr element = popoverData()->previouslyFocusedElement()) {
-        if (focusPreviousElement == FocusPreviousElement::Yes) {
+        if (focusPreviousElement == FocusPreviousElement::Yes && contains(document().focusedElement())) {
```

Comment:
This behavior change makes sense (e.g. you wouldn't want to restore focus if the person has tabbed away from the popover), but this should be in the HTML spec.

Reply:
https://github.com/whatwg/html/issues/9169

Reply:
Also we need to figure out if `contains()` is the correct call for popovers containing shadow DOM trees. (cc @rniwa)

Reply:
Agreed.
Original chromium commit:
https://chromium-review.googlesource.com/c/chromium/src/+/4322607

Reply:
> It might be 'containsIncludingShadowDOM'. @nt1m

Thanks, that could work, but I'll leave this for some other PR (also the relevant spec change still has not landed AFAIK).